### PR TITLE
[SPARK-48604][SQL] Replace deprecated `new ArrowType.Decimal(precision, scale)` method call

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This pr replaces deprecated classes and methods of `arrow-vector` called in Spark:

- `Decimal(int precision, int scale)` -> `Decimal(
      @JsonProperty("precision") int precision, 
      @JsonProperty("scale") int scale, 
      @JsonProperty("bitWidth") int bitWidth
    )`

All `arrow-vector` related Spark classes, I made a double check, only in `ArrowUtils` there is a deprecated  method call.
### Why are the changes needed?

Clean up deprecated API usage.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Passed GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by wayneguow** [comment]
cc @LuciferYang

File: sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala [hunk 1]
```diff
@@ -51,7 +51,7 @@ private[sql] object ArrowUtils {
     case BinaryType if !largeVarTypes => ArrowType.Binary.INSTANCE
     case _: StringType if largeVarTypes => ArrowType.LargeUtf8.INSTANCE
     case BinaryType if largeVarTypes => ArrowType.LargeBinary.INSTANCE
-    case DecimalType.Fixed(precision, scale) => new ArrowType.Decimal(precision, scale)
+    case DecimalType.Fixed(precision, scale) => new ArrowType.Decimal(precision, scale, 8 * 16)
```
Comment (LuciferYang) [inline sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala:54]: Can we use a named `val` to replace this magic number?

Comment (wayneguow) [reply sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala:54]: I refer to the style of lines 44-47

Comment (LuciferYang) [reply sql/api/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala:54]: OK

**Comment by wayneguow** [comment]
> Only one case?

Yes, i made a check.

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @wayneguow

**Pull request closed without merging**